ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/AvroDiffy.scala

```diff
@@ -18,118 +18,167 @@
-        getRawType(f.schema()).getType match {
-          case Schema.Type.RECORD =>
-            val a = x.flatMap(r => Option(r.get(name).asInstanceOf[GenericRecord]))
-            val b = y.flatMap(r => Option(r.get(name).asInstanceOf[GenericRecord]))
-            (a, b) match {
-              case (None, None)       => Nil
-              case (Some(_), None)    => Seq(Delta(fullName, a, None, UnknownDelta))
-              case (None, Some(_))    => Seq(Delta(fullName, None, b, UnknownDelta))
-              case (Some(_), Some(_)) => diff(a, b, fullName)
-            }
-          case Schema.Type.ARRAY if unordered.contains(fullName) =>
-            if (
-              unorderedFieldKeys.contains(fullName)
-              && isAvroRecordType(f.schema().getElementType)
-            ) {
-              val l = x
-                .flatMap(outer =>
-                  Option(outer.get(name).asInstanceOf[java.util.List[GenericRecord]].asScala.toList)
-                )
-                .getOrElse(List())
-                .flatMap(inner =>
-                  Try(inner.get(unorderedFieldKeys(fullName))).toOption.map(k => (k, inner))
-                )
-                .toMap
-              val r = y
-                .flatMap(outer =>
-                  Option(outer.get(name).asInstanceOf[java.util.List[GenericRecord]].asScala.toList)
-                )
-                .getOrElse(List())
-                .flatMap(inner =>
-                  Try(inner.get(unorderedFieldKeys(fullName))).toOption.map(k => (k, inner))
-                )
-                .toMap
-              (l.keySet ++ r.keySet).flatMap(k => diff(l.get(k), r.get(k), fullName)).toList
-            } else {
-              val a = x
-                .flatMap(r => Option(r.get(name).asInstanceOf[java.util.List[GenericRecord]]))
-                .map(sortList)
-              val b = y
-                .flatMap(r => Option(r.get(name).asInstanceOf[java.util.List[GenericRecord]]))
-                .map(sortList)
-              if (a == b) {
-                Nil
-              } else {
-                Seq(Delta(fullName, a, b, delta(a.orNull, b.orNull)))
-              }
-            }
-          case _ =>
-            val a = x.flatMap(r => Option(r.get(name)))
-            val b = y.flatMap(r => Option(r.get(name)))
-            if (a == b) Nil else Seq(Delta(fullName, a, b, delta(a.orNull, b.orNull)))
-        }
-      }
-      .filter(d => !ignore.contains(d.field))
+  private def isNumericType(`type`: Schema.Type): Boolean = `type` match {
+    case Schema.Type.INT | Schema.Type.LONG | Schema.Type.FLOAT | Schema.Type.DOUBLE => true
+    case _                                                                           => false
   }
 
-  private def getRawType(schema: Schema): Schema = {
-    schema.getType match {
+  private def numericValue(value: AnyRef): Double = value match {
+    case i: java.lang.Integer => i.toDouble
```

Comment:
Why `.toDouble` here?

Reply:
Because [numericDelta](https://github.com/spotify/ratatool/blob/51ec7981aa283b6ca07742474166666780afbcdc/ratatool-diffy/src/main/scala/com/spotify/ratatool/diffy/Diffy.scala#L118) only supports double